hydroflow/src/scheduled/state.rs

```diff
@@ -0,0 +1,18 @@
+use std::marker::PhantomData;
+
+use super::StateId;
+
+#[must_use]
+pub struct StateHandle<T> {
```

Comment:
Seems conceivable to me we want these to be refcounted so state that is no longer needed can be dropped, but I don't have a pattern in mind yet that would make use of that.